lib/benchee/benchmark.ex

```diff
@@ -43,9 +43,23 @@ defmodule Benchee.Benchmark do
   @spec measure(Suite.t, module) :: Suite.t
   def measure(suite = %Suite{jobs: jobs, configuration: config}, printer \\ Printer) do
     printer.configuration_information(suite)
-    run_times = record_runtimes(jobs, config, printer)
-
-    %Suite{suite | run_times: run_times}
+    results = record_results(jobs, config, printer)
+    run_times = parse_results(results, fn({key, {run_times, _}}) ->
+      {key, run_times}
+    end)
+    memory_usages = parse_results(results, fn({key, {_, memory_usages}}) ->
+      {key, memory_usages}
+    end)
```

Comment:
Line 98 to 101 already go like:

    job_run_times = results |> Enum.map(fn({run_times, _memory_usages}) -> run_times end) |> List.flatten
    memory_usages = results |> Enum.map(fn({_run_times, memory_usages}) -> memory_usages end) |> List.flatten
    {name, {job_run_times, memory_usages}}

It seems like we already separate memory usages and run times, and then here we do it again. It's the question if we could provide another data structure down there or maybe even 2 separate data structure so that we don't have to distinguish between memory/run times in 2 different places but have it localized in one spot.

Does that make more sense? :D O think "further down" wasn't exactly accurate in my first comment :)